lxd/cluster/recover.go

```diff
@@ -136,6 +136,39 @@ func updateLocalAddress(database *db.Node, address string) error {
 	return nil
 }
 
+// Create patch file for global nodes database.
+func writeGlobalNodesPatch(database *db.Node, nodes []db.RaftNode) error {
+	// No patch needed if there are no nodes
+	if len(nodes) < 1 {
+		return nil
+	}
+
+	var sql strings.Builder
+	for _, node := range nodes {
+		fmt.Fprintf(&sql, "UPDATE nodes SET address = %q WHERE id = %d;\n", node.Address, node.ID)
+	}
+
+	filePath := filepath.Join(database.Dir(), "patch.global.sql")
+	file, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+
+	defer func() { _ = file.Close() }()
```

Comment:
We should have a revert in this function to ensure if an error occurs the file gets deleted and isn't left in a partially written state.

Reply:
I guess custom patches are, by definition, not easy to forsee; I agree that we probably shouldn't append to one. My only concern is that we should avoid unnecessary failures late in the recovery process on the template node (when `writeGlobalNodesPatch` is called). I'll check for the presence of a patch file before modifying the dqlite state.